model/DataStore/PersistDataService.php

```diff
@@ -42,6 +43,16 @@ class PersistDataService extends ConfigurableService
     private const PACKAGE_FILENAME = 'QTIPackage';
     private const ZIP_EXTENSION = '.zip';
 
+    /** @var ExporterInterface */
+    private $exporter;
+
+    public function __construct($options = [], $exporter)
+    {
+        parent::__construct();
+        $this->setOptions($options);
+        $this->exporter = ($exporter instanceof ExporterInterface) ? $exporter : $this->getTestExporter();
```

Comment:
Instead of the constructor, don't you think this should come from the `$this->getOption(self::OPTION_EXPORTER_SERVICE_ID)` and you can use service locator to build it?

I am not sure overriding the constructor is good, cause we will be forcing pass options in 2 different places, which is the opposite of the current approach.

Wdyt?

Reply:
done